### PR TITLE
Fix SONiC compilation for bfnsde-9.0.0 and customize deployment on IMT devices.

### PR DESCRIPTION
**Pull request by YuriiHordynskyi**
**- Why I did it**
To fix SONiC compilation for bfnsde-9.0.0 and customize deployment on IMT devices.
**- How I did it**
Modified image build, install, and deploy scripts.
**- How to verify it**
https://docs.google.com/document/d/1apFjxSa7DJBwt6L0U0TGHC56aRnRtqOqmOXgSLa2DC8/edit#

```
Debian GNU/Linux 9 sonic ttyS0

sonic login: admin
Password: 
Linux sonic 4.9.0-9-2-amd64 #1 SMP Debian 4.9.168-1+deb9u3 (2015-12-19) x86_64
You are on
  ____   ___  _   _ _  ____
 / ___| / _ \| \ | (_)/ ___|
 \___ \| | | |  \| | | |
  ___) | |_| | |\  | | |___
 |____/ \___/|_| \_|_|\____|

-- Software for Open Networking in the Cloud --

Unauthorized access and/or use are prohibited.
All access and/or use are subject to monitoring.

Help:    http://azure.github.io/SONiC/

admin@sonic:~$ cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
VERSION_CODENAME=stretch
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
admin@sonic:~$ 

```

### CODE REVIEW

File: platform/barefoot/bfn-platform.mk [hunk 1]
```diff
@@ -1,5 +1,5 @@
-BFN_PLATFORM = bfnplatform_9.0.0.cc6ccbe_pr_deb9.deb
-$(BFN_PLATFORM)_URL = "https://github.com/barefootnetworks/sonic-release-pkgs/raw/rel_9_0/bfnplatform_9.0.0.cc6ccbe_pr_deb9.deb"
+BFN_PLATFORM = bfnplatform_1.0.0_amd64.deb
+$(BFN_PLATFORM)_URL = "http://192.168.157.83/yuriih/barefoot/bfnplatform_1.0.0_amd64.deb"
```
Comment (YuriiHordynskyi) [inline platform/barefoot/bfn-platform.mk]: Need to be changed to store the Barefoot related packages in some commonplace accessible for us and customers.

Comment (AndriiKit) [inline platform/barefoot/bfn-platform.mk]: Maybe it is better to use some general location and don't store package under your personal directory on 192.168.157.83 server

Comment (YuriiHordynskyi) [reply platform/barefoot/bfn-platform.mk]: Please see the comment above. I said the same =)

**Review by AndriiKit — APPROVED** [review]

File: build_debian.sh [hunk 2]
```diff
@@ -539,8 +543,8 @@ sudo mkdir -p $FILESYSTEM_ROOT/var/lib/docker
 sudo mksquashfs $FILESYSTEM_ROOT $FILESYSTEM_SQUASHFS -e boot -e var/lib/docker -e $PLATFORM_DIR
 
 ## Compress docker files
-pushd $FILESYSTEM_ROOT && sudo tar czf $OLDPWD/$FILESYSTEM_DOCKERFS -C ${DOCKERFS_PATH}var/lib/docker .; popd
+pushd $FILESYSTEM_ROOT && sudo env GZIP=-9 tar czf $OLDPWD/$FILESYSTEM_DOCKERFS -C ${DOCKERFS_PATH}var/lib/docker .; popd
```
Comment (vitalivanov) [inline build_debian.sh:546]: @YuriiHordynskyi why do we need that?

File: dockers/docker-base-stretch/Dockerfile.j2 [hunk 3]
```diff
@@ -74,10 +74,18 @@ RUN apt-get -y purge   \
 
 # Clean up apt
 # Remove /var/lib/apt/lists/*, could be obsoleted for derived images
-RUN apt-get clean -y                     && \
-    apt-get autoclean -y                 && \
-    apt-get autoremove -y                && \
-    rm -rf /var/lib/apt/lists/* /tmp/*
+RUN apt-get clean -y         && \
```
Comment (vitalivanov) [inline dockers/docker-base-stretch/Dockerfile.j2:77]: @YuriiHordynskyi you should submit it back to sonic as PR.

File: platform/barefoot/bfn-platform.mk [hunk 4]
```diff
@@ -1,5 +1,5 @@
-BFN_PLATFORM = bfnplatform_9.0.0.cc6ccbe_pr_deb9.deb
-$(BFN_PLATFORM)_URL = "https://github.com/barefootnetworks/sonic-release-pkgs/raw/rel_9_0/bfnplatform_9.0.0.cc6ccbe_pr_deb9.deb"
+BFN_PLATFORM = bfnplatform_1.0.0_amd64.deb
```
Comment (vitalivanov) [inline platform/barefoot/bfn-platform.mk:1]: we will correct this in next commits when server is ready.

File: build_debian.sh (hunk 2 above)
Comment (YuriiHordynskyi) [reply build_debian.sh:546]: To decrease image size more than 15MB.

File: rules/config [hunk 5]
```diff
@@ -36,7 +36,7 @@ SONIC_CONFIG_MAKE_JOBS = $(shell nproc)
 DEFAULT_USERNAME = admin
 
 # DEFAULT_PASSWORD - default password for installer build
-DEFAULT_PASSWORD = YourPaSsWoRd
+DEFAULT_PASSWORD = admin123
```
Comment (vitalivanov) [inline rules/config]: Let's have sonic password as default. It's sonic at the end...

File: rules/config [hunk 6]
```diff
@@ -77,7 +77,7 @@ ENABLE_ORGANIZATION_EXTENSIONS = y
 #SONIC_PROFILING_ON = y
 
 # ENABLE_SYSTEM_TELEMETRY - build docker-sonic-telemetry for system telemetry support
-ENABLE_SYSTEM_TELEMETRY = y
+ENABLE_SYSTEM_TELEMETRY = n
```
Comment (vitalivanov) [inline rules/config:80]: Why? because of size?

File: platform/barefoot/bfn-platform.mk (hunk 4 above)
Comment (YuriiHordynskyi) [reply platform/barefoot/bfn-platform.mk:1]: ok

File: rules/config (hunk 5 above)
Comment (YuriiHordynskyi) [reply rules/config]: ok will revert

File: rules/config (hunk 6 above)
Comment (YuriiHordynskyi) [reply rules/config:80]: barefoot 9.0.0 did not support telemetry and compilation was broken.

File: dockers/docker-base-stretch/Dockerfile.j2 (hunk 3 above)
Comment (YuriiHordynskyi) [reply dockers/docker-base-stretch/Dockerfile.j2:77]: ok. will create the PR if it's not present on the master branch.

**Review by vitalivanov — APPROVED** [review]

**Pull request merged**